homeassistant/components/climate/__init__.py

```diff
@@ -948,6 +990,54 @@ async def async_service_temperature_set(
         else:
             kwargs[value] = temp
 
+    if not hasattr(entity, "min_temperature_range"):
+        await entity.async_set_temperature(**kwargs)
+        return
+
+    if (
+        (min_temp_range := entity.min_temperature_range)
+        and (target_low_temp := kwargs.get(ATTR_TARGET_TEMP_LOW))
+        and (target_high_temp := kwargs.get(ATTR_TARGET_TEMP_HIGH))
+        and (target_high_temp - target_low_temp) < min_temp_range
+    ):
+        # Ensure target_low_temp is not higher than target_high_temp.
+        if target_low_temp > target_high_temp:
+            raise ServiceValidationError(
+                translation_domain=DOMAIN,
+                translation_key="low_temp_higher_than_high_temp",
+            )
```

Comment:
It's somewhat weird this is only checked if we have a min temperature range. Shouldn't this always be checked if we don't allow it? If base components don't check it, do integration check and handle it?

Reply:
Prel PR: https://github.com/home-assistant/core/pull/124488
Then we should move this out of this PR 👍